java/configuration/src/main/java/sleeper/configuration/properties/InstanceProperties.java

```diff
@@ -128,4 +131,13 @@ public static String tagsToString(Map<String, String> tags) {
         return builder.toString();
     }
 
+    public List<InstanceProperty> getAllGroupedProperties() {
+        List<InstanceProperty> allProperties = getAllProperties();
+        allProperties.sort(Comparator.comparingInt(p -> PropertyGroup.all().indexOf(p.getPropertyGroup())));
+        return allProperties;
+    }
+
+    public List<InstanceProperty> getAllProperties() {
+        return combineLists(List.of(UserDefinedInstanceProperty.values()), List.of(SystemDefinedInstanceProperty.values()));
+    }
```

Comment:
These methods are related to InstanceProperty rather than InstanceProperties. InstanceProperties is for the values rather than the actual properties.